frontend/components/homepage/dashboard.tsx

```diff
@@ -1,23 +1,40 @@
-import { Box, Center, Flex, Heading, ListItem, UnorderedList } from "@chakra-ui/react";
+import { Box, Button, Center, Flex, Heading, ListItem, UnorderedList, useDisclosure } from "@chakra-ui/react";
 import Card from "./card";
 import { links, link } from "@frontend/components/homepage/dashboard-structure";
 import { useSession } from "next-auth/react";
 import { USER_ROLES } from "@frontend/utils/constants";
 import DashboardCharts from "@frontend/components/homepage/dashboard-charts";
+import PatientInfoModal from "@frontend/components/modal/modal";
+import ConfirmAppointment from "@frontend/components/homepage/confirm-appointment";
+
+export interface AppointmentInfo {
+    AppointmentId: number;
+    Date: string;
+    Time: string;
+    Doctor_DoctorId: number;
+    Patient_PatientId: number;
+    Status: string;
+}
 
 export default function Dashboard({
     data,
     stats,
     statusFilled,
+    appointmentConfirmation,
+    incomingAppointments,
 }: {
     data: unknown[];
     stats: { unassignedPatientsCount: number; pendingCount: number };
     statusFilled: boolean;
+    appointmentConfirmation: AppointmentInfo[];
+    incomingAppointments: AppointmentInfo[];
```

Comment:
this seems to be getting pretty big, maybe creating an interface here could be nice